x-pack/plugins/cases/server/services/user_actions/helpers.ts

```diff
@@ -55,103 +119,43 @@ export const transformNewUserAction = ({
   owner,
 });
 
-interface BuildCaseUserAction {
-  action: UserAction;
-  actionAt: string;
-  actionBy: User;
-  caseId: string;
-  owner: string;
-  fields: UserActionField | unknown[];
-  newValue?: string | unknown;
-  oldValue?: string | unknown;
-  subCaseId?: string;
-}
+const createCaseReferences = (caseId: string, subCaseId?: string): SavedObjectReference[] => [
+  {
+    type: CASE_SAVED_OBJECT,
+    name: CASE_REF_NAME,
+    id: caseId,
+  },
+  ...(subCaseId
+    ? [
+        {
+          type: SUB_CASE_SAVED_OBJECT,
+          name: SUB_CASE_REF_NAME,
+          id: subCaseId,
+        },
+      ]
+    : []),
+];
 
-interface BuildCommentUserActionItem extends BuildCaseUserAction {
+interface BuildCommentUserActionItem extends BuildCaseUserActionParams {
   commentId: string;
 }
 
-export const buildCommentUserActionItem = ({
-  action,
-  actionAt,
-  actionBy,
-  caseId,
-  commentId,
-  fields,
-  newValue,
-  oldValue,
-  subCaseId,
-  owner,
-}: BuildCommentUserActionItem): UserActionItem => ({
-  attributes: transformNewUserAction({
-    actionField: fields as UserActionField,
-    action,
-    actionAt,
-    owner,
-    ...actionBy,
-    newValue: newValue as string,
-    oldValue: oldValue as string,
-  }),
-  references: [
-    {
-      type: CASE_SAVED_OBJECT,
-      name: `associated-${CASE_SAVED_OBJECT}`,
-      id: caseId,
-    },
-    {
-      type: CASE_COMMENT_SAVED_OBJECT,
-      name: `associated-${CASE_COMMENT_SAVED_OBJECT}`,
-      id: commentId,
-    },
-    ...(subCaseId
-      ? [
-          {
-            type: SUB_CASE_SAVED_OBJECT,
-            id: subCaseId,
-            name: `associated-${SUB_CASE_SAVED_OBJECT}`,
-          },
-        ]
-      : []),
-  ],
-});
+export const buildCommentUserActionItem = (params: BuildCommentUserActionItem): UserActionItem => {
```

Comment:
Dry'd this up to leverage the `buildCaseUserActionItem` since it was doing the same thing.